dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaClusterCPUChartTask.java

```diff
@@ -0,0 +1,153 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.ByteSink;
+import com.google.edwmigration.dumper.application.dumper.MetadataDumperUsageException;
+import com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.ClouderaManagerHandle.ClouderaClusterDTO;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.apache.hc.core5.net.URIBuilder;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The task collects hosts from Cloudera Manager <a
+ * href="https://cldr2-aw-dl-gateway.cldr2-cd.svye-dcxb.a5.cloudera.site/static/apidocs/resource_TimeSeriesResource.html">TimeSeries
+ * API</a> The chart is for whole cluster is available on {@code /cmf/home/} and {@code
+ * cmf/clusters/{clusterId}/status} pages in Cloudera Manager UI. <b/> The query to chart is written
+ * on <a
+ * href="https://docs.cloudera.com/documentation/enterprise/latest/topics/cm_dg_tsquery.html">tsquery</a>
+ * language.
+ */
+public class ClouderaClusterCPUChartTask extends AbstractClouderaManagerTask {
+  enum TimeSeriesAggregation {
+    RAW,
+    TEN_MINUTELY,
+    HOURLY,
+    SIX_HOURLY,
+    DAILY,
+    WEEKLY,
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(ClouderaCMFHostsTask.class);
+  /*
+    SELECT cpu_percent_across_hosts WHERE entityName = "1546336862" AND category = CLUSTER
+  */
+  private static final String TS_CPU_QUERY_TEMPLATE =
+      "SELECT cpu_percent_across_hosts WHERE entityName = \"%s\" AND category = CLUSTER";
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  private final int includedLastDays;
+  private final TimeSeriesAggregation tsAggregation;
+
+  public ClouderaClusterCPUChartTask() {
+    this(1, TimeSeriesAggregation.RAW);
+  }
+
+  public ClouderaClusterCPUChartTask(int includedLastDays, TimeSeriesAggregation tsAggregation) {
+    super(
+        String.format(
+            "cmf-cluster-cpu-%s-%s.jsonl",
```

Comment:
let's define some naming convention for charts, for example:
`cmf-cluster-cpu-{days}d-{agg}-chart.jsonl`
`cmf-cluster-ram-{days}d-{agg}-chart.jsonl`
`cmf-host-cpu-{days}d-{agg}-chart.jsonl`.

Do you have some other ideas?

Reply:
I'm not really sure about `{agg}` in filename, we have it file content and it seems we will able to use one loader for different aggregation ranges.

Reply:
I am concerned about scenario when we run several tasks for the same date range with different granularity (for ex 1D / 6H). If it's not an our case then I think we can remove `{agg}` part.

Reply:
Maybe we can extract the logic to compose chart name into a separate method to maintain consistency in the future.

Reply:
@dominik-klimczak do you mean to use utility method outside the scope of task class (to share it with another tasks)?

Reply:
I'm not sure about the scope. I don't know if there can be anything similar in other tasks. However, at least here, we could have a helper method to make the structure expressed explicitly as the result of a separate method.

Reply:
One line formatting is ok. The parameter are `final`, so we do not have consistency issues here.

Reply:
Added quick modification.